src/microservices/Microservices.IsIdentifiable/Runners/IsIdentifiableAbstractRunner.cs

```diff
@@ -219,14 +240,41 @@ public void LoadRules(string yaml)
 
         // ReSharper disable once UnusedMemberInSuper.Global
         public abstract int Run();
-
+        
         /// <summary>
         /// Returns each subsection of <paramref name="fieldValue"/> which violates validation rules (e.g. the CHI found).
         /// </summary>
         /// <param name="fieldName"></param>
         /// <param name="fieldValue"></param>
         /// <returns></returns>
         protected IEnumerable<FailurePart> Validate(string fieldName, string fieldValue)
+        {
+            // every time we lookup a value increment the counter and see if we should check cache overflow
+            if(_valuesLookedUp++ % _checkEvery == 0)
+            {
+                _valuesLookedUp = 0;
+                foreach (var c in Caches)
+                {
+                    //any column with more than the maximum number of cached values should be cleared
+                    if(c.Value.Keys.Count > MaxCacheSize)
+                        c.Value.Clear();
```

Comment:
I'm not sure that completely clearing all the cached values for a key is the best thing to do here, since we may lose values that are still useful. Would it not be better to drop the first added or least recently used? I assume that would require a more complex implementation though.

Reply:
I think simplicity wins out here, storing the durations and last used times and ordering by it probably costs more than just recaching a bunch of stuff.

Reply:
Changed to `MemoryCache` that should handle this implicitly